CMakeLists.txt

```diff
@@ -91,23 +137,49 @@ if(CUDA_FOUND)
     endif(CUDNN_FOUND)
   endif(USE_CUDNN)
 
-  if(USE_NCCL)
-    find_package(NCCL)
-    if(NCCL_FOUND)
-      include_directories(${NCCL_INCLUDE_DIR})
-      set(EXT_LIBS ${EXT_LIBS} ${NCCL_LIBRARIES})
-      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_NCCL")
-      LIST(APPEND CUDA_NVCC_FLAGS -DUSE_NCCL; )
-    endif(NCCL_FOUND)
-  endif(USE_NCCL)
-
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DCUDA_FOUND")
   list(APPEND CUDA_NVCC_FLAGS -DCUDA_FOUND; )
 
   if(MSVC)
     list(APPEND CUDA_NVCC_FLAGS -DBOOST_PP_VARIADICS=0; )
   endif()
 
+  if(USE_NCCL)
+    # define and set the include dir for the generated nccl.h header
+    set(NCCL_HEADER_LOCATION "${CMAKE_CURRENT_BINARY_DIR}/nccl/include")
+    include_directories(${NCCL_HEADER_LOCATION})
+
+    # set the path for the generated static lib
+    set(NCCL_STATIC "${CMAKE_CURRENT_BINARY_DIR}/nccl/lib/libnccl_static.a")
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -DUSE_NCCL")
+
+    LIST(APPEND CUDA_NVCC_FLAGS -DUSE_NCCL; )
+
+    # disables compilation for sm_30 to avoid ptxas warning... that's general Kepler support. But K80s are supported for instance by sm_35
```

Comment:
Does Marian still support Kepler? Does it actually work? I had a problem with back compat, where some CUDA operations on an old architecture would just compute zeroes, without getting an error.